lib/cloudflare_protected_email_address.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require 'cgi'
+
+class CloudflareProtectedEmailAddress
+  def initialize(obfuscated:)
+    @obfuscated = obfuscated
+  end
+
+  def human_readable
+    CGI.unescape(unobfuscated)
+  end
+
+  private
+
+  attr_accessor :obfuscated
+
+  def obfuscated_characters
+    obfuscated.scan(/.{2}/).drop(1).map(&:hex)
+  end
+
+  def key
+    @key ||= obfuscated[0..1].hex
```

Comment:
is this really expensive enough to need caching? (Especially compared to `unobfuscated`)